### PR TITLE
Retry more things

### PR DESCRIPTION
**Pull request by fw42**

### CODE REVIEW

File: turbopuffer/backend.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def make_api_request(self,
                 performance['request_time'] = time.monotonic() - request_start
                 # print(f'Request time (HTTP {response.status_code}):', performance['request_time'])
 
-                if response.status_code > 500:
+                if response.status_code > 500 or response.status_code == 408 or response.status_code == 429:
```
Comment (sirupsen) [inline turbopuffer/backend.py:97]: We should be retrying 502s etc., probably anything above 500?

Comment (fw42) [reply turbopuffer/backend.py:97]: Yep that's what the `> 500` here does

Comment (sirupsen) [reply turbopuffer/backend.py:97]: OK, so the other way around, we should retry 500s :)

Comment (fw42) [reply turbopuffer/backend.py:97]: You're saying you want to retry `>= 500`, not just `> 500`?

Comment (sirupsen) [reply turbopuffer/backend.py:97]: Yeah, don't you agree?

Comment (fw42) [reply turbopuffer/backend.py:97]: Sure, sounds good to me. Will change.

File: turbopuffer/backend.py [hunk 2]
```diff
@@ -140,14 +140,18 @@ def make_api_request(self,
...
-                    raise_api_error(http_err.response.status_code,
-                                   f'Request to {http_err.request.url} failed after {retry_attempt} attempts',
-                                   str(http_err))
+
+                    if isinstance(err, requests.HTTPError):
+                        raise_api_error(err.response.status_code,
+                                   f'Request to {err.request.url} failed after {retry_attempt} attempts',
+                                   str(err))
+                    else:
+                        raise
```
Comment (pushrax) [inline turbopuffer/backend.py:157]: is there any difference between `raise` and `raise err`?

File: turbopuffer/backend.py [hunk 3]
```diff
@@ -140,14 +140,18 @@ def make_api_request(self,
...
                     raise_api_error(response.status_code, 'Server returned non-JSON response', response.text)
-            except requests.HTTPError as http_err:
+            except (requests.HTTPError, requests.ConnectionError, requests.Timeout) as err:
                 retry_attempt += 1
                 # print(traceback.format_exc())
                 if retry_attempt < tpuf.max_retries:
                     # print(f'Retrying request in {2 ** retry_attempt}s...')
                     time.sleep(2 ** retry_attempt)  # exponential falloff up to 64 seconds for 6 retries.
                 else:
                     print(f'Request failed after {retry_attempt} attempts...')
```
Comment (pushrax) [inline turbopuffer/backend.py:150]: very unrelated: not a great practise to just print stuff in a library like this

**Review by pushrax — APPROVED** [review]

File: turbopuffer/backend.py (hunk 2 above)
Comment (fw42) [reply turbopuffer/backend.py:157]: Not 100% sure but I believe it's the same as in Ruby, i.e., `raise` lets the exception that was caught bubble up again while `raise e` raises a "new" exception and overwrites the backtrace that `e` had before.

**Pull request merged**